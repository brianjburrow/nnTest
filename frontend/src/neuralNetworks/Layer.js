import React from 'react';
import uuid from 'react-uuid';
import NewLayerForm from "./NewLayerForm";


function formatParams(params){
    let paramList = []
    for (const [key, value] of Object.entries(params)){
        paramList.push(<li key = {uuid()}>{key}:{value}</li>)
    }
    return <ul>{paramList}</ul>
}

function Layer({ type, params, layerIndex, insertLayer, deleteLayer }) {
    const tempParams = {...params}
    // Only the first LSTM layer can have a custom shape, so we only want
    // to show the first layer's input shape.  However, we don't want to
    // delete it from the data structure in case the user deletes the 
    // layer that is currently first.
    if (type === 'lstm' && layerIndex != 0){
        delete tempParams.inputShape
    }
    return (
        <div>
            Layer #{layerIndex}
            <p>{type}</p>
            {formatParams(tempParams)}
            <button onClick={insertLayer}>Insert Layer</button>
            <button onClick={deleteLayer}>Delete Layer</button>
        </div>
    )
}

export default Layer;