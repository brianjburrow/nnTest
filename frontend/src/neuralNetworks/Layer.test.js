// tests based on Jobly App (Springboard)

import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Layer from "./Layer";
import { UserProvider } from "../testUtils";

let TEST_LAYER = {
    type:'lstm', 
    params:{units:25, inputShape:'3'}, 
    layerIndex:1, 
    insertLayer:()=>{console.log('l')}, 
    deleteLayer:()=>{console.log('d')},
}

it("renders without crashing", function () {
    render(<Layer/>);
});

it("matches snapshot when logged in", function () {
    const { asFragment } = render(
        <MemoryRouter>
          <UserProvider>
            <Layer />
          </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function () {
    const { asFragment } = render(
        <MemoryRouter>
          <UserProvider currentUser={null}>
            <Layer />
          </UserProvider>
        </MemoryRouter>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
  