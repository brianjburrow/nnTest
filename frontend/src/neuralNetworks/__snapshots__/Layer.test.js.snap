// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot when logged in 1`] = `
<DocumentFragment>
  <div>
    Layer #
    <p />
    <ul />
    <button>
      Insert Layer
    </button>
    <button>
      Delete Layer
    </button>
  </div>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <div>
    Layer #
    <p />
    <ul />
    <button>
      Insert Layer
    </button>
    <button>
      Delete Layer
    </button>
  </div>
</DocumentFragment>
`;
